consular/main.py

```diff
@@ -457,30 +466,19 @@ def purge_dead_app_labels(self, apps):
             The list of apps as returned by the Marathon API.
         """
         # Get the existing keys
-        d = self.get_consul_consular_keys()
-        d.addErrback(self._ignore_not_found_error, [])
+        try:
+            keys = yield self.get_consul_consular_keys()
+        except UnexpectedResponseError as e:
+            if e.response.code == NOT_FOUND:
+                keys = []
+            else:
+                raise e
 
         # Filter the present apps out
-        d.addCallback(self._filter_marathon_apps, apps)
+        keys = self._filter_marathon_apps(keys, apps)
 
         # Delete the remaining keys
-        return d.addCallback(self.delete_consul_kv_keys, recurse=True)
-
-    def _ignore_not_found_error(self, failure, return_value=None):
```

Comment:
Basically, I don't trust that the callback/errback chain is working as I expect it to. It's a sanity check really.